core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -920,8 +908,23 @@ protected SearchService newSearchService(ClusterService clusterService, IndicesS
         return customNameResolvers;
     }
 
-    /** Constructs an internal node used as a client into a cluster fronted by this tribe node. */
-    protected Node newTribeClientNode(Settings settings, Collection<Class<? extends Plugin>> classpathPlugins, Path configPath) {
+    public static class NodeBuilder {
```

Comment:
do we really need this additional class? it's seem like users can provide all the stuff at once to the newNode method?